src/bridge/cockpitdisksamples.c

```diff
@@ -153,3 +158,134 @@ cockpit_disk_samples (CockpitSamples *samples)
   g_strfreev (lines);
   g_free (contents);
 }
+
+static gchar*
+read_file (const gchar *path)
+{
+  g_autoptr(GError) error = NULL;
+  gchar *contents = NULL;
+  if (!g_file_get_contents (path, &contents, NULL, &error))
+    {
+      /* Do not log:
+       * - file does not exist: process died and can no longer be sampled
+       * - access denied: bridge is not privileged and can't sample all processes
+       * - sometimes ERROR_FAILED is returned with message: "No such process"
+       */
+      if (g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT) ||
+          g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_ACCES) ||
+          (g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_FAILED) && g_strrstr(error->message, "No such process")))
+        {
+          return NULL;
+        }
+
+      g_warning ("Domain: %s, code: %d, message: %s", g_quark_to_string (error->domain), error->code, error->message);
+      return NULL;
+    }
+
+  return contents;
+}
+
+static void
+table_add_values(GHashTable *table,
+                   gchar *cgroup,
+                   guint64 read,
+                   guint64 write)
+{
+  cgroup_values_t *old = NULL;
+  if ((old = g_hash_table_lookup (table, cgroup)) != NULL)
+    {
+      // update values
+      old->read += read;
+      old->write += write;
+      g_free (cgroup);
+    }
+  else
+    {
+      // create new entry
+      cgroup_values_t *new = g_malloc0 (sizeof (cgroup_values_t));
+      new->read = read;
+      new->write = write;
+      g_hash_table_insert (table, cgroup, new);
+    }
+}
+
+static void
+get_process_io (const gchar *dir_name,
+                GHashTable *table)
+{
+  g_autofree gchar *path = g_strdup_printf ("/proc/%s/io", dir_name);
+  g_autofree gchar *contents = read_file (path);
+  if (!contents)
+    return;
+
+  guint64 read = 0, write = 0;
+  gchar **lines = g_strsplit (contents, "\n", -1);
+  for (guint n = 0; lines != NULL && lines[n] != NULL; n++)
+    {
+      gchar description[32] = {'\0'};
+      guint value;
+      const gchar *line = lines[n];
+      if (strlen (line) == 0)
+        continue;
+
+      if (sscanf (line, "%s %d", description, &value) != 2)
```

Comment:
I guess `%d` isn't going to be enough here.  You store the totals in 64bit fields, so probably this needs to be a 64bit read as well.